### PR TITLE
[20.09] treewide: mark broken for 20.09

### LINKED ISSUE #100111: Mark failing to build packages in 20.09 as broken
**Describe the bug**
As Per ZERO Hydra Failure #80379, we're supposed to mark every package that fails to build on the stable branch as broken. People can backport fixes to unbreak packages after we cut the final release. This is best done all at once semi-automatically, so this will be a task for @NixOS/nixos-release-managers  to perform when they are about to cut the final release. (in a PR please :exclamation: )

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
using git hub file changes to make sure the diff looks good

Fixes #100111

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/generic-builder.nix [hunk 1]
```diff
@@ -636,7 +636,8 @@ stdenv.mkDerivation ({
 
   };
 
-  meta = { inherit homepage license platforms; }
+  meta = {
+    broken = true; inherit homepage license platforms; }
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/generic-builder.nix:640]: This file is the builder for Haskell packages defined in nixpkgs (similar to `stdenv.mkDerivation` for normal packages).

It is probably a mistake to set `broken = true` here.

I imagine what happened is that there is some individual Haskell package that is not working and needs to be set as broken, but instead some sort of script just marked the general builder as broken instead.

Individual Haskell packages can be marked as broken in the file `pkgs/development/haskell-modules/hackage-packages.nix` on the release branches.  (Search for the string `broken` in this file to see what this looks like.)

Comment (jonringer) [reply pkgs/development/haskell-modules/generic-builder.nix]: yea, i noticed this, i forced pushed a change to the script im using

Comment (jonringer) [reply pkgs/development/haskell-modules/generic-builder.nix]: thanks for noticing though!

**Comment by cole-h** [comment]
cc issue https://github.com/NixOS/nixpkgs/issues/100111

**Comment by jonringer** [comment]
cc @worldofpeace should this close ZHF. This seems like the "capstone" for stabilization

**Comment by worldofpeace** [comment]
@jonringer Yes. People can still contribute fixes. Be sure to comment there as well.

**Comment by jonringer** [comment]
> @jonringer Yes. But people can still contribute fixes to unbreak things. Be sure to comment there as well.

Of course, but it would be weird to have the ZHF item open indefinitely. I'll just change the original post to mention that the backport commit may need to remove "broken = true;"

**Comment by worldofpeace** [comment]
> Of course, but it would be weird to have the ZHF item open indefinitely.

I was agreeing with "yes".

**Comment by jonringer** [comment]
Not sure why i read that as a rebuttable 

I guess I violently agree as well

**Comment by jonringer** [comment]
all that remains are the linux modules:
```
linuxPackages_5_8.batman_adv
linuxPackages_5_8.chipsec
linuxPackages_5_8.exfat-nofuse
linuxPackages_5_8.mxu11x0
linuxPackages_5_8.ndiswrapper
linuxPackages_5_8.netatop
linuxPackages_5_8.nvidiabl
linuxPackages_5_8.openafs_1_8
linuxPackages_5_8.r8168
linuxPackages_5_8.rtl8814au
linuxPackages_5_8.rtl8821au
linuxPackages_5_8.rtl88xxau-aircrack
linuxPackages_5_8.rtlwifi_new
linuxPackages_5_9.anbox
linuxPackages_5_9.batman_adv
linuxPackages_5_9.chipsec
linuxPackages_5_9.dpdk
linuxPackages_5_9.ena
linuxPackages_5_9.evdi
linuxPackages_5_9.exfat-nofuse
linuxPackages_5_9.mxu11x0
linuxPackages_5_9.ndiswrapper
linuxPackages_5_9.netatop
linuxPackages_5_9.nvidiabl
linuxPackages_5_9.openafs_1_8
linuxPackages_5_9.r8168
linuxPackages_5_9.rtl8814au
linuxPackages_5_9.rtl8821au
linuxPackages_5_9.rtl88xxau-aircrack
linuxPackages_5_9.rtlwifi_new
linuxPackages_5_9.virtualbox
linuxPackages_5_9.virtualboxGuestAdditions
linuxPackages_5_9.zfs
linuxPackages_5_9.zfsStable
linuxPackages_hardened.mxu11x0
linuxPackages_hardened.ndiswrapper
linuxPackages_hardened.openafs_1_8
linuxPackages_hardened.r8168
linuxPackages_hardened.rtl8814au
linuxPackages_hardened.rtl8821au
linuxPackages_hardened.rtlwifi_new
linuxPackages_latest-libre.anbox
linuxPackages_latest-libre.batman_adv
linuxPackages_latest-libre.chipsec
linuxPackages_latest-libre.exfat-nofuse
linuxPackages_latest-libre.mxu11x0
linuxPackages_latest-libre.ndiswrapper
linuxPackages_latest-libre.netatop
linuxPackages_latest-libre.nvidiabl
linuxPackages_latest-libre.openafs_1_8
linuxPackages_latest-libre.r8168
linuxPackages_latest-libre.rtl8814au
linuxPackages_latest-libre.rtl8821au
linuxPackages_latest-libre.rtl88xxau-aircrack
linuxPackages_latest-libre.rtlwifi_new
linuxPackages_latest.anbox
linuxPackages_latest.batman_adv
linuxPackages_latest.chipsec
linuxPackages_latest.exfat-nofuse
linuxPackages_latest.mxu11x0
linuxPackages_latest.ndiswrapper
linuxPackages_latest.netatop
linuxPackages_latest.nvidiabl
linuxPackages_latest.openafs_1_8
linuxPackages_latest.r8168
linuxPackages_latest.rtl8814au
linuxPackages_latest.rtl8821au
linuxPackages_latest.rtl88xxau-aircrack
linuxPackages_latest.rtlwifi_new
linuxPackages_latest_hardened.anbox
linuxPackages_latest_hardened.batman_adv
linuxPackages_latest_hardened.chipsec
linuxPackages_latest_hardened.exfat-nofuse
linuxPackages_latest_hardened.mxu11x0
linuxPackages_latest_hardened.ndiswrapper
linuxPackages_latest_hardened.netatop
linuxPackages_latest_hardened.nvidiabl
linuxPackages_latest_hardened.openafs_1_8
linuxPackages_latest_hardened.r8168
linuxPackages_latest_hardened.rtl8814au
linuxPackages_latest_hardened.rtl8821au
linuxPackages_latest_hardened.rtl88xxau-aircrack
linuxPackages_latest_hardened.rtlwifi_new
linuxPackages_latest_xen_dom0.anbox
linuxPackages_latest_xen_dom0.batman_adv
linuxPackages_latest_xen_dom0.chipsec
linuxPackages_latest_xen_dom0.exfat-nofuse
linuxPackages_latest_xen_dom0.mxu11x0
linuxPackages_latest_xen_dom0.ndiswrapper
linuxPackages_latest_xen_dom0.netatop
linuxPackages_latest_xen_dom0.nvidiabl
linuxPackages_latest_xen_dom0.openafs_1_8
linuxPackages_latest_xen_dom0.r8168
linuxPackages_latest_xen_dom0.rtl8814au
linuxPackages_latest_xen_dom0.rtl8821au
linuxPackages_latest_xen_dom0.rtl88xxau-aircrack
linuxPackages_latest_xen_dom0.rtlwifi_new
linuxPackages_latest_xen_dom0_hardened.anbox
linuxPackages_latest_xen_dom0_hardened.batman_adv
linuxPackages_latest_xen_dom0_hardened.chipsec
linuxPackages_latest_xen_dom0_hardened.exfat-nofuse
linuxPackages_latest_xen_dom0_hardened.mxu11x0
linuxPackages_latest_xen_dom0_hardened.ndiswrapper
linuxPackages_latest_xen_dom0_hardened.netatop
linuxPackages_latest_xen_dom0_hardened.nvidiabl
linuxPackages_latest_xen_dom0_hardened.openafs_1_8
linuxPackages_latest_xen_dom0_hardened.r8168
linuxPackages_latest_xen_dom0_hardened.rtl8814au
linuxPackages_latest_xen_dom0_hardened.rtl8821au
linuxPackages_latest_xen_dom0_hardened.rtl88xxau-aircrack
linuxPackages_latest_xen_dom0_hardened.rtlwifi_new
linuxPackages_testing_bcachefs.ndiswrapper
linuxPackages_testing_bcachefs.rtl8814au
linuxPackages_testing_bcachefs.rtl8821au
linuxPackages_xen_dom0.mxu11x0
linuxPackages_xen_dom0.ndiswrapper
linuxPackages_xen_dom0.r8168
linuxPackages_xen_dom0.rtl8814au
linuxPackages_xen_dom0.rtl8821au
linuxPackages_xen_dom0.rtlwifi_new
linuxPackages_xen_dom0_hardened.mxu11x0
linuxPackages_xen_dom0_hardened.ndiswrapper
linuxPackages_xen_dom0_hardened.openafs_1_8
linuxPackages_xen_dom0_hardened.r8168
linuxPackages_xen_dom0_hardened.rtl8814au
linuxPackages_xen_dom0_hardened.rtl8821au
linuxPackages_xen_dom0_hardened.rtlwifi_new
linuxPackages_zen.anbox
linuxPackages_zen.batman_adv
linuxPackages_zen.chipsec
linuxPackages_zen.exfat-nofuse
linuxPackages_zen.mxu11x0
linuxPackages_zen.ndiswrapper
linuxPackages_zen.netatop
linuxPackages_zen.nvidiabl
linuxPackages_zen.openafs_1_8
linuxPackages_zen.r8168
linuxPackages_zen.rtl8814au
linuxPackages_zen.rtl8821au
linuxPackages_zen.rtl88xxau-aircrack
linuxPackages_zen.rtlwifi_new
```

**Comment by jonringer** [comment]
I will probably tackle the linux  modules in a different PR, as we don't really have a good way (that I'm aware of) to determine the "flavor" of the kernel (e.g. libre, zen, xen, hardened, etc.)

**Comment by worldofpeace** [comment]
That's an awful lot of modules.

**Comment by worldofpeace** [comment]
I hope for people's hardware they can't rebuild switch because a module for the stable kernel doesn't build. Obviously something to handle after release.

**Comment by jonringer** [comment]
```
https://github.com/NixOS/nixpkgs/pull/101736
2 packages marked as broken and skipped:
python37Packages.pythonmagick python38Packages.pythonmagick

4 packages built:
python27Packages.locustio python27Packages.msal-extensions python27Packages.python-axolotl python37Packages.sphinxcontrib-spelling
```

**Comment by jonringer** [comment]
going to "forward port" some of the python fixes in a different PR

**Pull request merged**